sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/WindowTest.java

```diff
@@ -220,6 +224,67 @@ public void testMissingLateness() {
       .apply("Trigger", Window.<String>triggering(trigger));
   }
 
+  @Test
+  @Category(RunnableOnService.class)
+  public void testNoWindowFnDoesNotReassignWindows() {
+    pipeline.enableAbandonedNodeEnforcement(true);
+
+    PCollection<Long> initialWindows =
+        pipeline
+            .apply(CountingInput.upTo(10L))
+            .apply(
+                "AssignInitialTimestamps",
+                WithTimestamps.of(
+                    new SerializableFunction<Long, Instant>() {
+                      @Override
+                      public Instant apply(Long input) {
+                        return new Instant(input);
+                      }
+                    }))
+            .apply("AssignWindows", Window.<Long>into(FixedWindows.of(Duration.millis(5L))));
+
+    PAssert.that(initialWindows)
+        .inWindow(new IntervalWindow(new Instant(0L), new Instant(5L)))
+        .containsInAnyOrder(0L, 1L, 2L, 3L, 4L);
+    PAssert.that(initialWindows)
+        .inWindow(new IntervalWindow(new Instant(5L), new Instant(10L)))
+        .containsInAnyOrder(5L, 6L, 7L, 8L, 9L);
+
+    PCollection<Long> newTimestamps =
+        initialWindows.apply(
+            "AssignSkewedTimestamps",
+            WithTimestamps.of(
+                new SerializableFunction<Long, Instant>() {
+                  @Override
+                  public Instant apply(Long input) {
+                    return new Instant(input + 10L);
+                  }
+                }));
+    PAssert.that(newTimestamps)
+        .inWindow(new IntervalWindow(new Instant(0L), new Instant(5L)))
+        .containsInAnyOrder(0L, 1L, 2L, 3L, 4L);
+    PAssert.that(newTimestamps)
+        .inWindow(new IntervalWindow(new Instant(5L), new Instant(10L)))
+        .containsInAnyOrder(5L, 6L, 7L, 8L, 9L);
+
+    // Should in general only update the windowing strategy, not any windows that were assigned
```

Comment:
Comment seems a bit in the weeds. Maybe just "The elements should be in the same windows, though their new timestamps would not be assigned to those windows, if window assignment were re-run."